drivers/include/sdp3x.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2019 Dirk Ehmen
+ *               2020 Nishchay Agrawal
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sdp3x SDP3x temperature and differential pressure sensor
+ * @ingroup     drivers_sensors
+ * @{
+ * @file
+ * @brief       Device driver interface for the SDP3x sensor.
+ *
+ * @author      Dirk Ehmen       <ehmen@ibr.cs.tu-bs.de>
+ * @author      Nishchay Agrawal <f2016088@pilani.bits-pilani.ac.in>
+ * @}
+ */
+
+#ifndef SDP3X_H
+#define SDP3X_H
+
+#include "saul.h"
+#include "xtimer.h"
+#include "mutex.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ *  Flags for sensor reading
+ */
+#define SDP3X_FLAG_MASS_FLOW  0x02
+#define SDP3X_FLAG_DIFF_PRESS 0x00
+#define SDP3X_FLAG_AVERAGE    0x01
+#define SDP3X_FLAG_NO_AVERAGE 0x00
+
+/**
+ * @brief   SDP3x models
+ */
+typedef enum {
+    SDP3x_31 = 1,
+    SDP3x_32 = 2
+} sdp3x_model_t;
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                  /**< I2C device which is used */
+    uint8_t i2c_addr;               /**< I2C address */
+    sdp3x_model_t model;            /**< Model Number */
+    gpio_t irq_pin;                 /**< IRQ pin to check data ready */
+} sdp3x_params_t;
+
+/**
+ * @brief   Device descriptor for the SDP3x sensor
+ */
+typedef struct {
+    bool continuous_measurement;    /**< Device measuring data continuously */
+    sdp3x_params_t params;          /**< Device initialization parameters */
+    mutex_t mutex;                  /**< Mutex for this sdp3x dev */
+} sdp3x_t;
+
+/**
+ * @brief   Measurement from SDP3x sensor
+ */
+typedef struct {
+    int32_t differential_pressure;  /**< Differential Pressure measured 
+                                         in centiPa */
+    int32_t temperature;            /**< Temperature measured in 100*°C */
+    int16_t dp_scale_factor;       /**< Scale factor for measurement */
```

Comment:
You can drop this now - it's only relevant inside the driver and otherwise unused.

Reply:
Resolved